### PR TITLE
Fix an issue with MapLayerPreferences that would not save with the proper key with replacing an existing source.

### PR DESCRIPTION
**Pull request by mdastous-bentley**
.. also added `MapLayersPrefBrowserStorage` as a simple/testing implementation of `UserPreferencesAccess`.

### CODE REVIEW

**Comment by mdastous-bentley** [comment]
@calebmshafer we might want to move `PreferencesBrowserStorage` in a more "common" package if you think it might be useful for other applications.

**Comment by mdastous-bentley** [comment]
@aruniverse please review

File: packages/itwin/map-layers/src/PreferencesBrowserStorage.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+* @beta
+*/
+export interface PreferencesBrowserStorageOptions {
+  /** indicate whether or not delete function should throw an error if the provided key cannot be found. */
+  throwOnDeleteMissingKey?: boolean;
+}
+
+/** User preferences implementation using browser's local storage.
+  * It creates storage item key in the form of:
+  *     <storageItemKeyPrefix>.<project_guid?>.<imodel_guid?>
```
Comment (aruniverse) [inline packages/itwin/map-layers/src/PreferencesBrowserStorage.ts:25]: we're not using the term project anymore, can you replace with itwin?

Comment (aruniverse) [reply packages/itwin/map-layers/src/PreferencesBrowserStorage.ts:25]: L27 too

**Review by aruniverse — APPROVED** [review]

**Comment by mdastous-bentley** [comment]
@eringram please review

File: packages/itwin/map-layers/src/PreferencesBrowserStorage.ts (hunk 1 above)
Comment (mdastous-bentley) [reply packages/itwin/map-layers/src/PreferencesBrowserStorage.ts]: @aruniverse should be good now...

**Review by eringram — APPROVED** [review]

**Pull request merged**